### DISCUSSION TITLE
Fields array, useForm state resetting to default when removing a field : is that a bug ?

### DISCUSSION THREAD

**Discussion by SimTheFool** [react-hook-form/react-hook-form#4644] [Q&A] [answered]
Hey !

I am facing a behaviour I doesn't really understand, and I would need some help to understand it, and resolve it. I'm not really at ease with React, and totally new to react hook form, which doesn't help. Here is some codesandbox : https://codesandbox.io/s/test-field-array-ewrvt?file=/src/index.js

I'm having some issue with the submitted data.

I've a group of inputs (firstname and lastname in the example) controlled with a controller. I'm using fields array to use several times those inputs, and they are populated with the default value given to useForm hook.

When I remove one of the controller of the fields array (changing the number of controlled inputs rendered by the app), everything seems ok. But if I submit directly after removing, we can see that we have the truncated default values array. If, instead of submitting directly, I trigger some change in the controlled inputs, the changed input is saved, but the other are still reset to their default value.

I would like/need my code to react properly to  the removal of a fields array element, and I doesn't know how to proceed.

Edit / NB : 

Also, is there a way to use the default id generated by the fields array without it being submitted with the data ?

**Comment by SimTheFool** [comment, marked as answer]
Ok, I just migrate to v7 and everything seem to work fine for now, sorry for that ^^'

I'll mark this topic as anwered later, because the behaviour I talked about is still present in v6 and just in case I figure I'm still having issues with v7.

**Reply by bluebill1049** [reply]
what's the issue in v7? let me know

**Reply by SimTheFool** [reply]
Everything seems to work fine for v7 so far, the only issue was for v6. so that's all good for me ^^

**Reply by bluebill1049** [reply]
sweet, awesome to hear.
